path/file.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package path
+
+import "os"
+
+// Exists checks if specified file, directory, or symlink target exists. If
+// filename is a symlink the symlink will be followed and Exists reports the
+// existence of the symlink target.
+func Exists(filename string) (bool, error) {
```

Comment:
The comments are helpful now, thanks. But I wonder if we should change the function names to help readers? I definitely wouldn't have guessed at the symlink behavior based on these names. I think a carefully named parameter could make this almost self-documenting? Here's a rough draft of what I'm thinking. Sorry for making this an ordeal.

```
type LinkTreatment int
const (
  CheckSymlinkTarget LinkTreatment = iota
  CheckSymlink
  // I can imagine other behaviors, like recursively checking the target, or rejecting symlinks entirely...
)

// ...
func Exists(follow LinkTreatment, filename string) (bool, error) ...
```

Reply:
(feel free to argue with me about whether this is worth changing or not, btw)